### PR TITLE
Feature/update to polkadot v0 8 29

### PR DESCRIPTION
**Pull request by MRamanenkau**
[DDC pallet](https://github.com/Cerebellum-Network/ddc-pallet/pull/6)

### CODE REVIEW

File: frame/ddc-metrics-offchain-worker/src/tests/test_runtime.rs [hunk 1]
```diff
@@ -167,6 +167,7 @@ parameter_types! {
     pub const SurchargeReward: Balance = 150;
     pub const MaxDepth: u32 = 100;
     pub const MaxValueSize: u32 = 16_384;
+    pub MaxCodeSize: u32 = 256 * 1024;
```
Comment (MRamanenkau) [inline frame/ddc-metrics-offchain-worker/src/tests/test_runtime.rs:170]: This value is higher than we have in node runtime. Probably we need to increase it there to be able to upload a SC

File: bin/node/runtime/src/lib.rs [hunk 2]
```diff
@@ -1028,12 +1072,13 @@ construct_runtime!(
...
 		System: frame_system::{Module, Call, Config, Storage, Event<T>},
 		Utility: pallet_utility::{Module, Call, Event},
-		Babe: pallet_babe::{Module, Call, Storage, Config, Inherent, ValidateUnsigned},
+		Babe: pallet_babe::{Module, Call, Storage, Config, ValidateUnsigned},
 		Timestamp: pallet_timestamp::{Module, Call, Storage, Inherent},
 		Authorship: pallet_authorship::{Module, Call, Storage, Inherent},
 		Indices: pallet_indices::{Module, Call, Storage, Config<T>, Event<T>},
 		Balances: pallet_balances::{Module, Call, Storage, Config<T>, Event<T>},
 		TransactionPayment: pallet_transaction_payment::{Module, Storage},
+		ElectionProviderMultiPhase: pallet_election_provider_multi_phase::{Module, Call, Storage, Event<T>, ValidateUnsigned},
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:1082]: [pallet_election_provider_multi_phase](https://crates.parity.io/pallet_election_provider_multi_phase/index.html)
[Original PR](https://github.com/paritytech/substrate/pull/7319)

File: bin/node/runtime/src/lib.rs [hunk 3]
```diff
@@ -738,6 +779,7 @@ parameter_types! {
 			<Runtime as pallet_contracts::Config>::WeightInfo::on_initialize_per_queue_item(1) -
 			<Runtime as pallet_contracts::Config>::WeightInfo::on_initialize_per_queue_item(0)
 		)) / 5) as u32;
+	pub MaxCodeSize: u32 = 128 * 1024;
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs]: “All entry points pre-charge the weight for accessing a maximum size code and correct this weight after execution.”
[link to the PR description](https://github.com/paritytech/substrate/pull/8086)

File: bin/node/runtime/src/lib.rs [hunk 4]
```diff
@@ -1306,15 +1358,24 @@ impl_runtime_apis! {
 		}
 	}
 
+	#[cfg(feature = "try-runtime")]
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:1362]: [Testing tool](https://docs.substrate.io/reference/command-line-tools/try-runtime/#:~:text=The%20try%2Druntime%20tool%20is,state%20before%20going%20to%20production.)

File: bin/node/runtime/src/lib.rs [hunk 5]
```diff
@@ -738,6 +779,8 @@ parameter_types! {
 			<Runtime as pallet_contracts::Config>::WeightInfo::on_initialize_per_queue_item(1) -
 			<Runtime as pallet_contracts::Config>::WeightInfo::on_initialize_per_queue_item(0)
 		)) / 5) as u32;
+	// Make it possible to upload ddc.wasm
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/src/lib.rs:782]: We need to get rid of ddc in the comment

File: bin/node/runtime/src/lib.rs (hunk 2 above)
Comment (AndreiNavoichyk) [reply bin/node/runtime/src/lib.rs:1082]: In what version Polkadot did add it?

File: bin/node/runtime/src/lib.rs [hunk 6]
```diff
@@ -738,6 +779,8 @@ parameter_types! {
...
+	pub MaxCodeSize: u32 = 160 * 1024;
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/src/lib.rs:783]: Let's get rid of reference to ddc.wasm

**Review by AndreiNavoichyk — COMMENTED** [review]
@MRamanenkau 
Please, attach a PR to the ddc-pallet.

File: bin/node/runtime/src/lib.rs (hunk 2 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:1082]: In v0.8.29

File: bin/node/runtime/src/lib.rs (hunk 5 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:782]: I wanted to make it clear why the value (160Kb) is chosen

Comment (AndreiNavoichyk) [reply bin/node/runtime/src/lib.rs:782]: Sure. But be more abstract.
For example, "Set to 160KB to support Cere's internal SCs"

**Review by AndreiNavoichyk — DISMISSED** [review]

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**